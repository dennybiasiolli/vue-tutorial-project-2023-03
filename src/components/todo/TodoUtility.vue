<script setup>
import { computed, onMounted, ref } from 'vue'
import TodoList from './TodoList.vue'

const items = ref([])
function filterTodoCondition(item) {
  return !item.completed
}
// const todoItems = computed(() => items.value.filter(item => !item.completed))
const todoItems = computed(() => items.value.filter(filterTodoCondition))
const completedItems = computed(() => items.value.filter(item => item.completed))

function handleSwitchTodo({ id, completed }) {
  // function searchItem(item) {
  //   return item.id === id
  // }
  // const item = items.value.find(searchItem)
  const item = items.value.find(item => item.id === id)
  item.completed = completed
}

onMounted(() => {
  items.value = [
    { id: 1, text: 'farina', completed: false },
    { id: 2, text: 'olio', completed: true },
    { id: 3, text: 'pane', completed: false },
    { id: 4, text: 'zucchero', completed: true },
    { id: 5, text: 'sale', completed: false },
  ]
})
</script>

<template>
  <h2>To-do Utility</h2>

  <TodoList
    title="To-do list"
    :items="todoItems"
    @shitchCompletedTodo="handleSwitchTodo"
  />
  <TodoList
    title="Completed list"
    :items="completedItems"
    @shitchCompletedTodo="handleSwitchTodo"
  />
</template>
