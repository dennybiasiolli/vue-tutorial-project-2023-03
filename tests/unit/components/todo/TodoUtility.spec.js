import { createTestingPinia } from '@pinia/testing'
import { shallowMount } from '@vue/test-utils'
import { expect, it, vi } from 'vitest'
import TodoUtility from '@/components/todo/TodoUtility.vue'

function myShallowMount() {
  return shallowMount(TodoUtility, {
    global: {
      plugins: [createTestingPinia({
        createSpy: vi.fn,
        initialState: {
          todo: {
            items: [
              { id: 1, text: 'item 1', completed: false },
              { id: 2, text: 'item 2', completed: false },          
              { id: 3, text: 'item 3', completed: true },          
            ]
          },
        },
      })],
    },
  })
}

it('should render as expected', () => {
  const wrapper = myShallowMount()
  expect(wrapper.html()).toMatchSnapshot()
})
