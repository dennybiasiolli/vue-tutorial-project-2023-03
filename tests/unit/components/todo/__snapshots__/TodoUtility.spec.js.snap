// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render as expected 1`] = `
"<h2>To-do Utility</h2>
<form><input type=\\"text\\" placeholder=\\"Add to-do item\\"><button type=\\"submit\\" disabled=\\"\\">Add</button></form>
<todo-list-stub title=\\"To-do list\\" items=\\"[object Object],[object Object]\\"></todo-list-stub>
<todo-list-stub title=\\"Completed list\\" items=\\"[object Object]\\"></todo-list-stub>"
`;
